libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp

```diff
@@ -0,0 +1,573 @@
+    0x0a, 0x71, 0x08, 0xd5, 0x42, 0x21, 0xbd, 0x91, 0xbc, 0xd2, 0x42, 0xc1, 0x42, 0x00, 0xb2, 0xfe, 0x82, 0x05, 0x2f,
+    0x50, 0x30, 0x21, 0x2e, 0x21, 0xf2, 0x00, 0x2e, 0x00, 0x2e, 0xd0, 0x2e, 0xf0, 0x6f, 0x02, 0x30, 0x02, 0x42, 0x20,
+    0x26, 0xe0, 0x6f, 0x02, 0x31, 0x03, 0x40, 0x9a, 0x0a, 0x02, 0x42, 0xf0, 0x37, 0x05, 0x2e, 0x5e, 0xf7, 0x10, 0x08,
+    0x12, 0x24, 0x1e, 0xf2, 0x80, 0x42, 0x83, 0x84, 0xf1, 0x7f, 0x0a, 0x25, 0x13, 0x30, 0x83, 0x42, 0x3b, 0x82, 0xf0,
+    0x6f, 0x00, 0x2e, 0x00, 0x2e, 0xd0, 0x2e, 0x12, 0x40, 0x52, 0x42, 0x00, 0x2e, 0x12, 0x40, 0x52, 0x42, 0x3e, 0x84,
+    0x00, 0x40, 0x40, 0x42, 0x7e, 0x82, 0xe1, 0x7f, 0xf2, 0x7f, 0x98, 0x2e, 0x6a, 0xd6, 0x21, 0x30, 0x23, 0x2e, 0x61,
+    0xf5, 0xeb, 0x2c, 0xe1, 0x6f
+};
+/*
+ * End of Bosch microcode copyright
+ */
+
+#define BMI270_CHIP_ID 0x24
+#define BMI270_CMD_SOFTRESET 0xB6
+#define BMI270_CMD_FIFOFLUSH 0xB0
+#define BMI270_FIFO_ACC_EN 0x40
+#define BMI270_FIFO_GYR_EN 0x80
+#define BMI270_BACKEND_SAMPLE_RATE 1600
+
+/* Datasheet says that the device powers up and soft resets in 2s, so waiting for
+ * 5ms before initialization is enough. */
+#define BMI270_POWERUP_DELAY_MSEC 5
+
+/* This number of samples should provide only one read burst operation on the
+ * FIFO most of the time (99.99%). */
+#define BMI270_MAX_FIFO_SAMPLES 8
+
+#define BMI270_HARDWARE_INIT_MAX_TRIES 5
+
+extern const AP_HAL::HAL& hal;
+
+AP_InertialSensor_BMI270::AP_InertialSensor_BMI270(AP_InertialSensor &imu,
+                                                   AP_HAL::OwnPtr<AP_HAL::Device> dev,
+                                                   enum Rotation rotation)
+    : AP_InertialSensor_Backend(imu)
+    , _dev(std::move(dev))
+    , _rotation(rotation)
+{
+}
+
+AP_InertialSensor_Backend *
+AP_InertialSensor_BMI270::probe(AP_InertialSensor &imu,
+                                AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev,
+                                enum Rotation rotation)
+{
+    if (!dev) {
+        return nullptr;
+    }
+    auto sensor = new AP_InertialSensor_BMI270(imu, std::move(dev), rotation);
+
+    if (!sensor) {
+        return nullptr;
+    }
+
+    if (!sensor->_init()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+AP_InertialSensor_Backend *
+AP_InertialSensor_BMI270::probe(AP_InertialSensor &imu,
+                                AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev,
+                                enum Rotation rotation)
+{
+    if (!dev) {
+        return nullptr;
+    }
+    auto sensor = new AP_InertialSensor_BMI270(imu, std::move(dev), rotation);
+
+    if (!sensor) {
+        return nullptr;
+    }
+
+    if (!sensor->_init()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+void AP_InertialSensor_BMI270::start()
+{
+    _dev->get_semaphore()->take_blocking();
+
+    if (!_configure_accel()) {
```

Comment:
make the 3 configure fns void and removed the panic calls, it just wastes flash space, as they always return true

Reply:
Fixed